docs/developers/v2/hacking-with-yearn.md

```diff
@@ -0,0 +1,237 @@
+- [Brownie](https://eth-brownie.readthedocs.io/en/stable/)
+- [ethers.js](https://docs.ethers.io/v5/)
+- [Web3.js](https://web3js.readthedocs.io/en/v1.4.0/)
+- [Ganache](https://trufflesuite.com/ganache/) 
+- [Foundry](https://github.com/nicolasgarcia214/damn-vulnerable-defi-foundry)
+
+# Yearn Vaults & Smart Contract Integrations
+Yearn’s core product is the Vault (also known as yVault) — a set-it-and-forget-it yield aggregator running on top of DeFi’s yield-generating protocols like Compound, Aave, Curve, and Convex. Most of the core functionality such as deposit, withdraw, borrow, repay, or flashloan occurs in the XX contract ().
+For a breakdown of V2 features see the contract overview or dive even deeper with the V2 Technical Paper.
+
+## Getting Started
+
+#### Starter Kit
+#### 
+Here's an example for integrating V2 vaults and ERC-4626. 
+
+https://github.com/storming0x/ystarter-foundry-kit
+
+What you'll find:
+- Basic example Solidity Smart Contracts for integrating with Yearn Vaults for both V2 and ERC4626 interfaces.
+- ERC4626 adapter to wrap yearn vaults.
+- Example Contracts
+- Sample test suite.
+
+
+Yearn Vaults
+
+[This repository](https://github.com/yearn/yearn-vaults#readme) includes the set of smart contracts that are used for the Yearn Vaults. It contains the requirements, code, deployment scripts, and tests necessary for the core protocol, including a inheritable template strategy for use with Solidity-based strategies that interact with Yearn Vaults. These contracts are used to create a simple way to generate high risk-adjusted returns for depositors of various assets via best- in-class lending protocols, liquidity pools, and community-made yield farming strategies on Ethereum.
+
+
+#### 
+## Testing
+
+### **Introduction**
+You are able to test Yearn vaults without spending cryptoassets  on the available testnet for each network. 
+
+To get started, head to the yearn testnet client which is available at: https://hack.yearn.finance/
+
+Make sure to switch your wallet provider to the appropriate testnet network in order to use the protocol testnet without incurring costs on the mainnet network. 
+ 
+Testnet networks are different environments from mainnet and simulate real use cases for development, testing or production. The assets on a testnet are not “real,” meaning that they cannot be withdrawn to your wallet for any real economic value. 
+ 
+Testing is possible through Ethereum Rinkeby  
+
+**Steps**:
+- deploy your vault with your own token
+- mint your own tokens 
+- hack!
+
+
+### What do I need for testing?
+### 
+When accessing each testnet, you will need to use its own native asset token. 
+ 
+Steps to test: 
+
+- Go to https://hack.yearn.finance/
+- Switch to the testnet you wish to utilize over your wallet provider 
+- Make sure to have the native asset for the specific network 
+- Get some tokens from the test client faucet  
+Ethereum Rinkeby: https://faucet.rinkeby.io/  
+
```

Comment:
no bold on "Steps to test" as in the other bullets, there's a space between the bullets that was not in the previous ones